tools/cli/src/mentat_cli/repl.rs

```diff
@@ -198,10 +207,51 @@ impl Repl {
         }
     }
 
-    fn read_file(&self, files: Vec<String>) {
+    fn read_files(&mut self, files: Vec<String>) {
         for file in files {
-            println!("Executing edn in file {}", file);
+            let res = self.read_file(file);
+            if res.is_err() {
+                match res.unwrap_err() {
+                    cli::Error(err, _) => { println!("{}", err) },
+                }
+            }
+        }
+    }
+
+    fn read_file(&mut self, file: String) -> Result<(), cli::Error> {
+        let path = Path::new(&file);
+        let display = path.display();
+
+        let f = match File::open(&path) {
+            Err(err) => bail!(cli::ErrorKind::FileError(display.to_string(), err.to_string())),
+            Ok(file) => file,
+        };
+
+        let mut buffer = String::new();
+        let mut cmd_err: Option<edn::ParseError> = None;
+
+        let file = BufReader::new(&f);
```

Comment:
You have the whole file, correct?  Why not just ask for the input you really want: a vector of transactions, which is really a vector-of-vectors-of-assertions.  Then you can just use EDN to parse the whole thing and not monkey with this incremental parsing at all.  That is, accept:
```edn
[[[:db/add ...]
   [:db/remove]]
 [{:db/add ...}
  {:db/add ...}]]
```